cw-rct201/src/Components/Admin/AdminSingleProduct.tsx

```diff
@@ -1,518 +0,0 @@
-import {
-  Avatar,
-  Box,
-  Button,
-  Flex,
-  Heading,
-  Input,
-  SimpleGrid,
-  Text,
-  useDisclosure,
-} from "@chakra-ui/react";
-import axios from "axios";
-import React from "react";
-import {
-  Drawer,
-  DrawerBody,
-  DrawerFooter,
-  DrawerHeader,
-  DrawerOverlay,
-  DrawerContent,
-  DrawerCloseButton,
-} from "@chakra-ui/react";
-import {
-  AlertDialog,
-  AlertDialogBody,
-  AlertDialogFooter,
-  AlertDialogHeader,
-  AlertDialogContent,
-  AlertDialogOverlay,
-} from "@chakra-ui/react";
-import {
-  Menu,
-  MenuButton,
-  MenuList,
-  MenuItem,
-  MenuItemOption,
-  MenuGroup,
-  MenuOptionGroup,
-  MenuDivider,
-} from "@chakra-ui/react";
-import { Data } from "../../Pages/MobileProducts";
-import { useDispatch } from "react-redux";
-import { sendDataToRedux, updateTodoFromDom } from "../../Redux/admin/adminAction";
-import { Dispatch } from "redux";
-import { store } from "../../Redux/store";
-import { DeleteIcon, EditIcon } from "@chakra-ui/icons";
-import { toast } from "react-toastify";
-
-const AdminSingleProduct = ({ image1, id, name, cost, category }: Data) => {
-  const dispatch: any = useDispatch();
-
-  const [nameDrawerOpen, setNameDrawerOpen] = React.useState<boolean>(false);
-  const [priceDrawerOpen, setPriceDrawerOpen] = React.useState<boolean>(false);
-  const [updateOpen, setUpdateOpen] = React.useState<boolean>(false);
-  const [alertDeleteOpen, setDeleteOpen] = React.useState<boolean>(false);
-  const [newName, setnewName] = React.useState<string>("");
-  const [newCat, setNewCat] = React.useState<string>("");
-  const [newPrice, setNewPrice] = React.useState<any>();
-  const [categoryDrawerOpen, setCategoryDrawerOpen] = React.useState<boolean>(false);
-
-  const [newImage, setNewImage] = React.useState<string>("");
-  const [image1DrawerOpen, setImage1DrawerOpen] = React.useState<boolean>(false);
-
-  const cancelRef = React.useRef<any>();
-  const nameSuccess=()=>{
-    toast.success(" Name Updated Successfully",{theme:"colored"})
-}
-const categorySuccess=()=>{
-  toast.success(" Category Updated Successfully",{theme:"colored"})
-}
-const priceSuccess=()=>{
-  toast.success("Price Updated Successfully",{theme:"colored"})
-}
-const deleteSuccess=()=>{
-  toast.error("Product Deleted Successfully",{theme:"colored"})
-}
-const imageUpdate=()=>{
-  toast.success("Image Updated Successfully",{theme:"colored"})
-}
-  const updateName = (id: number) => {
-    let obj = {
-      name: newName,
-    };
-
-    axios.patch(`https://backendsirver-for-daily-needs.vercel.app/products/${id}`, obj);
-
-    setUpdateOpen(false);
-    setNameDrawerOpen(false);
-    setTimeout(()=>{
-      // dispatch(sendDataToRedux());
-    },1500)
-    
-   
-    nameSuccess()
-  };
-  const updatePrice = (id: number) => {
-    let obj = {
-      cost: +newPrice,
```

Comment:
why are you sending +newPrice in response? what is it determining? please add comments above this line